pkg/collector/metadata/agentchecks.go

```diff
@@ -20,7 +21,7 @@ type AgentChecksCollector struct{}
 // Send collects the data needed and submits the payload
 func (hp *AgentChecksCollector) Send(s *serializer.Serializer) error {
 	payload := agentchecks.GetPayload()
-	if err := s.SendMetadata(payload); err != nil {
+	if err := s.SendMetadata(payload, forwarder.TransactionPriorityNormal); err != nil {
```

Comment:
Why are some metadata high prio and some normal ? Shouldn't all metadata have priority over the rest ?

I'm not a fan that the sender taking a forwarder constant as a parameter. The sender was supposed to abstract all the complexity of sending a specific type of data. I would rather have `SendMetadata` and `SendHostMetadata` than having the caller decide how the forwarder will retry the transaction (unless we want all metadata to be high prio).

Reply:
> Why are some metadata high prio and some normal ? Shouldn't all metadata have priority over the rest ?

If the host is not known by the backend and the backend has an outage, the metrics received by the backends before the host metadata payload are not tags correctly. The host metadata payload is more important than the other metadata payload.

> The sender was supposed to abstract all the complexity of sending a specific type of data. I would rather have SendMetadata and SendHostMetadata

After a second though, I misunderstood the responsibility of `MetricSerializer` because of the generic type `marshaler.Marshaler`. I will update the code.